core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -443,18 +479,117 @@ private SearchResult parseToSearchResult(@Nonnull final IMAPResponse[] ir) throw
             }
             final List<Long> v = new ArrayList<Long>(); // will always return a non-null array
 
+            Long modSeq = null;
+
             // Grab all SEARCH responses
             long num;
             for (final IMAPResponse sr : ir) {
                 // There *will* be one SEARCH response.
                 if (sr.keyEquals("SEARCH")) {
                     while ((num = sr.readLong()) != -1) {
-                        v.add(Long.valueOf(num));
+                        v.add(num);
+                    }
+                    if (sr.readByte() == '(') {
+                        final String s = sr.readAtom();
+                        if (s.equalsIgnoreCase("MODSEQ")) {
+                            modSeq = sr.readLong();
+                        }
                     }
                 }
             }
 
-            return new SearchResult(v);
+            return new SearchResult(v, modSeq);
```

Comment:
modSeq might be null here, but calling a constructor with modSeq tagged as @Nonnull
Either fix the constructor or call the appropriate constructor